src/time.cpp

```diff
@@ -96,15 +96,21 @@ namespace g3 {
          return rv;
       }
       // error mode. just return sec, microsecond
-      ts->tv_sec  = now.tv_sec;
+      ts->tv_sec = now.tv_sec;
       ts->tv_nsec = now.tv_usec * 1000;
       return 0;
 #elif(defined(WIN32) || defined(_WIN32) || defined(__WIN32__))
-	   return timespec_get(ts, TIME_UTC);
+      static auto os =
+         std::chrono::time_point_cast<std::chrono::nanoseconds>(std::chrono::system_clock::now()).time_since_epoch() -
```

Comment:
I think it is good to still keep it within this wrapped ```timespec_get``` function though as it speaks volumes what it does by re-using the name.